File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_412_A.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var n, k int
	fmt.Scan(&n, &k)
	
	var v string
	fmt.Scan(&v)

	d := []string{"LEFT", "RIGHT"}
	f := k-1 < n-k
	m := min(k-1, n-k)
	
	var a []string
	for i := 0; i < m; i++ {
		a = append(a, d[1-f])
	}
	
	vArray := strings.Split(v, "")
	if !f {
		for i := len(vArray) - 1; i >= 0; i-- {
			a = append(a, "PRINT "+vArray[i])
			a = append(a, d[f])
		}
	} else {
		for _, i := range vArray {
			a = append(a, "PRINT "+i)
			a = append(a, d[f])
		}
	}
	
	for i := 0; i < len(a)-1; i++ {
		fmt.Println(a[i])
	}
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_412_A.java
import java.util.Scanner;

public class codeforces_412_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        sc.nextLine();
        String v = sc.nextLine();
        String[] d = {"LEFT", "RIGHT"};
        boolean f = k - 1 < n - k;
        int m = Math.min(k - 1, n - k);
        StringBuilder a = new StringBuilder();

        for (int i = 0; i < m; i++) {
            a.append(d[f ? 0 : 1]).append("\n");
        }

        for (int i : new int[]{v.length() - (f ? 1 : 0), -1, f ? 1 : -1}) {
            a.append("PRINT ").append(v.charAt(i)).append("\n");
            a.append(d[f ? 1 : 0]).append("\n");
        }

        System.out.print(a.substring(0, a.length() - 1));
    }
}
// End of Code
